File: VB/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Diagnostics

Namespace RepPrintOnDotMatrix

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim report As XtraReport1 = New XtraReport1()
            report.CreateDocument()
            printControl1.PrintingSystem = report.PrintingSystem
        End Sub

        Private Sub barButtonItem2_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            printControl1.PrintingSystem.ExportToCsv(Application.StartupPath & "\temporary.csv", New DevExpress.XtraPrinting.CsvExportOptions(",", Encoding.Default))
            Dim psi As ProcessStartInfo = New ProcessStartInfo()
            psi.FileName = Application.StartupPath & "\temporary.csv"
            psi.Verb = "Print"
            Call Process.Start(psi)
        End Sub
    End Class
End Namespace

File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;

namespace RepPrintOnDotMatrix {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            XtraReport1 report = new XtraReport1();

            report.CreateDocument();
            printControl1.PrintingSystem = report.PrintingSystem;
        }

        private void barButtonItem2_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            printControl1.PrintingSystem.ExportToCsv(Application.StartupPath + "\\temporary.csv", new DevExpress.XtraPrinting.CsvExportOptions(",", Encoding.Default));

            ProcessStartInfo startInfo = new ProcessStartInfo(Application.StartupPath + "\\temporary.csv");
            startInfo.Verb = "Open";

            foreach (var verb in startInfo.Verbs) {
                if (verb == "Print") startInfo.Verb = "Print";
            }
            Process.Start(startInfo);
        }

    }
}